index.js

```diff
@@ -1,68 +1,122 @@
 'use strict';
 
-/**
- * @usage
- *
- * gulp.src('app.js', {read: false})
- * 		 .pipe(rollup(options))
- * 		 .pipe(gulp.dest('dist'));
- */
-
-var through     = require('through2'),
-    gutil       = require('gulp-util'),
-    PluginError = gutil.PluginError,
-    fs          = require('fs'),
-    path        = require('path'),
-    PLUGIN_NAME = 'gulp-rollup';
+var PLUGIN_NAME = 'gulp-rollup';
+
+var util         = require('util');
+var gutil        = require('gulp-util');
+var PluginError  = gutil.PluginError;
+var File         = gutil.File;
+var Transform    = require('readable-stream').Transform;
+var hypothetical = require('rollup-plugin-hypothetical');
+var path         = require('path');
+
+function GulpRollup(options) {
+  Transform.call(this, { objectMode: true });
 
-module.exports = function(options) {
   options = options || {};
 
-  return through.obj(function(file, enc, callback) {
-    if (!file.path) { return callback(); }
+  var wonderland = {};
+  var entry, haveSourcemaps;
+
+  this._transform = function(file, enc, cb) {
+    if (!file.isBuffer()) {
+      this.emit('error', new PluginError(PLUGIN_NAME, 'Input files must be buffered!'));
+      return cb();
+    }
+
+    if (haveSourcemaps === undefined) {
+      haveSourcemaps = file.sourceMap !== undefined;
+    } else if (haveSourcemaps !== (file.sourceMap !== undefined)) {
+      this.emit('error', new PluginError(PLUGIN_NAME, 'Mixing of sourcemapped and non-sourcemapped files!'));
+      return cb();
+    }
+
+    if (haveSourcemaps) {
+      wonderland[file.path] = {
+        code: file.contents.toString(),
+        map:  file.sourceMap
+      };
+    } else {
+      wonderland[file.path] = file.contents.toString();
+    }
 
-    if (file.isStream()) {
-      return callback(new PluginError(PLUGIN_NAME, 'Streaming not supported'));
+    if (!entry && options.entry && hypothetical({ files: wonderland, allowRealFiles: true }).resolveId(options.entry)) {
```

Comment:
gulp-rollup is a changed package, @operandom. It swore never to go back to its old ways. So no using gulp.src as a globber. I will make it possible to have multiple entry files, though you'll have to use another package to do the globbing.